### DISCUSSION TITLE
Yolo v8 showing 1.00 confidence on literally every (non-existent) object on an image

### DISCUSSION THREAD

**Discussion by Olwar** [ultralytics/ultralytics#15009] [Q&A]
Hi!

The title explains it. Yolo v8 showing 1.00 confidence on literally every COCO object on a single image and annotates them.

This is the code I have right now:
```
from ultralytics import YOLO
import cv2

# Load the YOLOv8 model
model = YOLO('yolov8l.pt')  # Using the large model

# Load the image
image_path = 'bird.jpg'
image = cv2.imread(image_path)

if image is None:
    print(f"Error: Unable to load image from {image_path}")
else:
    # Get image dimensions
    height, width = image.shape[:2]
    print(f"Original image dimensions: {width}x{height}")

    # Run inference on the image with a larger size
    results = model(image, imgsz=1280)  # Set size to 1280, maintaining aspect ratio

    # Print all detections
    print("All detections:")
    for r in results:
        for box in r.boxes:
            class_id = int(box.cls)
            class_name = model.names[class_id]
            confidence = box.conf.item()
            x1, y1, x2, y2 = box.xyxy[0].tolist()
            print(f"Detected: {class_name} with confidence {confidence:.2f} at coordinates: ({x1:.2f}, {y1:.2f}, {x2:.2f}, {y2:.2f})")

    # Visualize the results on the image
    annotated_image = results[0].plot()

    # Display the annotated image
    cv2.imshow('YOLOv8 Detection', annotated_image)
    cv2.waitKey(0)
    cv2.destroyAllWindows()

    # Save the annotated image
    cv2.imwrite('bird_annotated_large.jpg', annotated_image)
    print("Annotated image saved as 'bird_annotated_large.jpg'")
```
    
    
    
And for some reason, the image looks like this annotated:    
![bird_annotated_large](https://github.com/user-attachments/assets/88f65d59-994e-45d8-9ae3-03c08bf9aeaa)

I also tried with different size models, same problem. Also tried with different images and webcam feed, same problem.

I have not finetuned the model, it's straight out of the box. The original image's dimensions are 1280x960.

**Comment by pderrenger** [comment]
@Olwar hi! Thanks for reaching out. It seems like you're encountering an unusual issue with the YOLOv8 model. First, please ensure that you're using the latest version of the Ultralytics package, as updates often include important bug fixes and improvements.

If the issue persists, it might be related to the model's confidence threshold settings. You can try adjusting the confidence threshold to filter out low-confidence detections. For example, setting a higher confidence threshold can help reduce false positives.

If the problem continues, please share more details about your environment, such as the specific version of the Ultralytics package and any additional configurations you might have applied. This information will help us diagnose the issue more effectively.

Feel free to check the [Ultralytics documentation](https://docs.ultralytics.com) for more insights and troubleshooting tips. Thank you!
